cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -177,14 +263,14 @@ def __init__(
         self._shared_storage_mount_dirs = shared_storage_mount_dirs
         self._shared_storage_attributes = shared_storage_attributes
 
-        self._add_resources()
+        self._add_resources(head_eni)
 
     @property
     def stack_name(self):
         """Name of the CFN stack."""
         return Stack.of(self.nested_stack_parent).stack_name
 
-    def _add_resources(self):
+    def _add_resources(self, head_eni):
```

Comment:
`head_eni` is already an instance variable (an was assigned a value). We can use `self.head_eni`.

Reply:
In this case it is not. It is a member of `class Pool(Construct)`.

However I can make it an object variable (in case we need also elsewhere) and use self._head_eni.